### DISCUSSION TITLE
Error boundaries for components

### DISCUSSION THREAD

**Discussion by Marcdj-02** [remix-run/remix#5917] [Q&A]
I'd like to catch errors within a component and render an fallback UI.

I understand that Remix had ErrorBoundaries for routes, but I could not find anything for wrapping components.
In the React documentation they talk about a ErrorBoundary component that catched rendering errors and renders an alternative component.
(see: https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary)

This implementation does not work with Remix, since it is SSR.
Other SSR-suported error boundaries did not work either.

Does anyone have a working ErrorBoundary for components for Remix and maybe it could be nice to make this built-in?

Example:
![afbeelding](https://user-images.githubusercontent.com/72502650/227802752-e461a32d-5925-408c-b1d9-f865205e7d17.png)

**Comment by lili21** [comment]
You can wrap it with [ClientOnly](https://github.com/sergiodxa/remix-utils#clientonly). render the part only in client.

**Reply by paul-vd** [reply]
So, this means for any component that that is wrapped in a `ComponentErrorBoundary` will no longer have SSR?
